rust/agents/validator/src/submit.rs

```diff
@@ -36,18 +36,18 @@ impl ValidatorSubmitter {
         let span = info_span!("ValidatorSubmitter");
         let reorg_period = Some(self.reorg_period);
         tokio::spawn(async move {
-            let mut current_index = self.checkpoint_syncer.latest_index().await?.unwrap_or_default();
+            let mut maybe_current_index = self.checkpoint_syncer.latest_index().await?;
             loop {
                 sleep(Duration::from_secs(self.interval)).await;
 
                 // Check the current checkpoint
                 let checkpoint = self.outbox.latest_checkpoint(reorg_period).await?;
 
-                if current_index < checkpoint.index {
+                if maybe_current_index.map_or(true, |current_index| current_index < checkpoint.index ) && !checkpoint.root.is_zero() {
```

Comment:
This took me a little to wrap my head around, is this equivalent?
```
if maybe_current_index.map_or(!checkpoint.root.is_zero(), |current_index| current_index < checkpoint.index ) {
```
Assuming I didn't misunderstand, I felt like this was easier for me to understand, but happy to defer to you on which one is more clear for you